core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java

```diff
@@ -1584,6 +1589,74 @@ ResultHandle createValue(MethodContext creator, MethodCreator method, ResultHand
         }
     }
 
+    final class NewRecorder extends DeferredArrayStoreParameter {
+        final Class<?> theClass;
+        final Constructor<?> injectCtor;
+
+        NewRecorder(Class<?> theClass) {
+            this.theClass = theClass;
+            Constructor<?> injectCtor = null;
+            Constructor<?>[] ctors = theClass.getDeclaredConstructors();
+            if (ctors.length == 1) {
+                injectCtor = ctors[0];
+            } else {
+                for (var i : ctors) {
+                    if (i.isAnnotationPresent(Inject.class)) {
+                        if (injectCtor == null) {
+                            injectCtor = i;
+                        } else {
+                            throw new RuntimeException("Multiple @Inject constructors on " + theClass);
+                        }
+                    }
+                }
+                if (injectCtor == null) {
+                    throw new RuntimeException(
+                            "Could not determine constructor for " + theClass + " add @Inject to a constructor");
+                }
+            }
+            this.injectCtor = injectCtor;
+        }
+
+        @Override
+        void doPrepare(MethodContext context) {
+            super.doPrepare(context);
+        }
+
+        @Override
+        ResultHandle createValue(MethodContext context, MethodCreator method, ResultHandle array) {
+            List<DeferredParameter> deferredParameters = new ArrayList<>();
+            if (injectCtor != null) {
+                try {
+                    for (var param : injectCtor.getParameterTypes()) {
+                        var obj = configCreatorFunction.apply(param);
+                        var result = findLoaded(obj);
+                        if (result == null) {
+                            throw new RuntimeException("Cannot inject object of type " + param);
+                        }
+                        deferredParameters.add(result);
+                    }
+                } catch (Exception e) {
+                    throw new RuntimeException("Failed to prepare injector constructor " + injectCtor + " of bytecode recorder",
+                            e);
+                }
+            }
+            if (injectCtor == null) {
+                return method.newInstance(ofConstructor(theClass));
+            } else {
+                try {
+                    List<ResultHandle> handles = new ArrayList<>();
+                    for (var result : deferredParameters) {
+                        handles.add(context.loadDeferred(result));
+                    }
+                    return method.newInstance(ofConstructor(injectCtor.getDeclaringClass(), injectCtor.getParameterTypes()),
+                            handles.toArray(ResultHandle[]::new));
```

Comment:
You can avoid an extra lambda class by using
```suggestion
                            handles.toArray(new ResultHandle[0]));
```

Reply:
We don't care about method references in deployment side code, they are everywhere